libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -1752,15 +1788,21 @@ MQTTStatus_t MQTT_SerializePublish( const MQTTPublishInfo_t * pPublishInfo,
     }
     else if( ( pPublishInfo->qos != MQTTQoS0 ) && ( packetId == 0U ) )
     {
-        LogError( ( "Packet Id is 0 for PUBLISH with QoS=%u.",
+        LogError( ( "Packet ID is 0 for PUBLISH with QoS=%u.",
+                    pPublishInfo->qos ) );
+        status = MQTTBadParameter;
+    }
+    else if( ( pPublishInfo->dup ) && ( pPublishInfo->qos == MQTTQoS0 ) )
+    {
+        LogError( ( "Duplicate flag is set for PUBLISH with Qos=%u,",
                     pPublishInfo->qos ) );
```

Comment:
Nit: This and similar lines will always be QoS 0:
```suggestion
        LogError( ( "Duplicate flag is set for PUBLISH with QoS 0." ) );
```

Reply:
thanks